Python/datacube.py

```diff
@@ -0,0 +1,71 @@
+import glob
+
+import numpy as np
+import xarray as xr
+import netCDF4 as nc
+
+def read_data(data_path, dv_key_name):
+    """Load an ensemble of CanRCM4
+    models into a single datacube.
+    ------------------------------
+    Args:
+        data_path (Str): path containing CanRCM4 ensemble
+        dv_key_name (Str): name of the design value as 
+            in the netCDF file
+    Returns:
+        ds (xarray Dataset): data cube of assembled ensemble models 
+            into a single variable
+    """
+    if not isinstance(data_path, str):
+        raise TypeError("data_path must be a string, got {}"
+                        .format(type(data_path)))
+
+    if not isinstance(dv_key_name, str):
+        raise TypeError("dv_key_name must be a string, got {}"
+                        .format(type(dv_key_name)))
+
+    # Create a list of all files in PATH
+    nc_list = np.asarray(glob.glob(data_path+"*.nc"))
+
+    # create an example NetCDF4 dataset
+    inst = nc.Dataset(nc_list[0], 'r')
+    data_cube = np.ones((inst['lat'].shape[0],
+                         inst['lat'].shape[1],
+                         nc_list.shape[0]))
+
+    inst.close()
+
+    # iterate through files in path
+    # to create data cube
+    for i, path in enumerate(nc_list):
+        run = nc.Dataset(path, 'r')
+        dv = run.variables[dv_key_name][:, :]
+        data_cube[:, :, i] = dv
+
+    lat = run.variables['lat'][:, :]
```

Comment:
So I'm having a bit of a hard time understand what you are doing here.  You have `run` set inside the loop and then later you use it (the last run from the loop) to set some variables up.  Did you want to only use the last `run` from `nc_list`?  If that is the case you may want to comment why you are doing so.  Furthermore, it may be a good idea to check that `run` got set properly in the loop before using it outside.
```
    run = None
    for i, path in enumerate(nc_list):
        run = nc.Dataset(path, 'r')
        dv = run.variables[dv_key_name][:, :]
        data_cube[:, :, i] = dv

    if run:
        # set vars
```
That's just what I would do, the others may differ in opinion here.